src/daft-io/src/object_io.rs

```diff
@@ -52,12 +52,27 @@ impl GetResult {
     }
 }
 
-#[derive(Debug)]
+#[derive(Debug, Clone, PartialEq)]
 pub enum FileType {
     File,
     Directory,
 }
-#[derive(Debug)]
+
+impl From<std::fs::FileType> for FileType {
+    fn from(value: std::fs::FileType) -> Self {
+        if value.is_dir() {
+            Self::Directory
+        } else if value.is_file() {
+            Self::File
+        } else if value.is_symlink() {
+            panic!("Should never encounter symlinks here, {:?}", value)
```

Comment:
Let change this to a `TryFrom` so we can propagate the error to the user.

Reply:
I started with a `TryFrom` implementation, but neither this `panic!()` nor the `unreachable!()` should happen unless there's an internal bug, so a `panic!()` seemed more appropriate so we could get the full stacktrace from the user.

But I'll revert it back to `TryFrom` + a new internal `DaftError` variant, if we think that's better.